pnut.c

```diff
@@ -434,16 +405,7 @@ void begin_string() {
   hash = 0;
 }
 
-void accum_string() {
-  hash = (ch + (hash ^ HASH_PARAM)) % HASH_PRIME;
-  string_pool[string_pool_alloc] = ch;
-  string_pool_alloc += 1;
-  if (string_pool_alloc >= STRING_POOL_SIZE) {
-    fatal_error("string pool overflow");
-  }
-}
-
-// Like accum_string, but takes the character as input instead of reading it from ch
+// Append a character to the current string under construction in the pool
```

Comment:
The reason to have `accum_string` in addition to `accum_string_char` is that `accum_string` is much faster when compiled to shell since it doesn't have any local variable or parameter.

Reply:
Cherry-picking the commit removing `accum_string` and running `./run-pnut-variant.sh --tokenizer`, it adds between 5 and 9% to the execution time. Not that significant, but on zsh it can mean a few extra seconds.

Time when tokenizing pnut-sh.c with `accum_string`:

```
Running tokenizer with dash
       18.14 real        15.72 user         2.17 sys
Running tokenizer with ksh
       12.84 real        12.32 user         0.42 sys
Running tokenizer with bash
       36.24 real        35.50 user         0.55 sys
```

With `accum_string_char`:
```
Running tokenizer with dash
       17.29 real        15.10 user         2.05 sys
Running tokenizer with ksh
       12.13 real        11.45 user         0.49 sys
Running tokenizer with bash
       34.10 real        33.20 user         0.61 sys
```

I don't have times for zsh on pnut-sh.c as it's painfully slow but on `examples/repl.c` we can also see the difference, going from 53.9s to 58.5s.

Reply:
Ok.  Do you want me to back this change out, then?

Reply:
Yes please. Once that's done and CI passes, we can merge :)

Reply:
Maybe the comment should say something about this performance issue.

Reply:
I should probably write a guide that goes over the shell-specific optimizations since it applies to more than just this function.